### PR TITLE
fix(www): sidebar bullets indention for steps

### LINKED ISSUE #24093: Tutorial step titles show a bullet point over number in dark mode
## Description

While in dark mode while viewing the docs. A noticeable circular bullet point covers up the number of the tutorial step. (Note: this also occurs on light mode, though the bullet is much fainter and doesn't seem to cause any issue seeing the number).

image

This is only an issue in the tutorial section, docs, features, contributing, are all unaffected.

## Related changes

This was introduced somewhere in #23248 which cleaned up a lot of code in the docs sidebar.

## How to fix

- [ ] identify what styles were adjusted that caused this (maybe as simple as a `list-style: none` thing, or maybe a custom `::before` element) and fix those styles (you can run the website following [instructions in the contributing section](https://www.gatsbyjs.org/contributing/setting-up-your-local-dev-environment/))
- [ ] open a pull request stating the changes made that references this issue like `Fixes #24093`

### PR DESCRIPTION
**Pull request by Pyrax**
## Description

Fixes indentation of sidebar bullets when parent link has `ui: steps`-property (i.e. in Tutorials section). For this I used a new prop called `parentUi` which is set on items.

Result looks like this:
Screenshot from 2020-05-15 22-56-20

## Related Issues

Fixes #24093

### CODE REVIEW

**Comment by tesseralis** [comment]
Hi @pyrax! Thank you for your contribution and the quick turnaround.

Unfortunately, this isn't the proper solution: instead of indenting, we don't want the bullets to show up *at all*, as shown here in Light Mode:

Screen Shot 2020-05-15 at 3 12 49 PM

You can do this with the `list-style` css property: https://developer.mozilla.org/en-US/docs/Web/CSS/list-style

Feel free to update this PR with the proper solution. Since this is a pretty obvious bug, I might handle this myself since I have context about these files.

**Comment by tesseralis** [comment]
Sorry for the earlier confusing comment! This is indeed the right approach, but here is how I would implement it:

1. Create a prop called `isSteps` on the `list-item` component
2. In [`accordion.js`](https://github.com/gatsbyjs/gatsby/blob/master/www/src/components/sidebar/accordion.js#L66), change the call to the following:

```jsx
          {item.items.map(subitem => (
            <Item item={subitem} key={subitem.title} isSteps={item.ui === `steps`} />
          ))}
```

That way the child item doesn't need to have the unnecessary information about its parent.

**Comment by tesseralis** [comment]
In addition, I noticed that in dark mode, the bullets show up as white (the foreground color), while in light mode, they still show up as white (the background color). @fk, is this intentional? What's the expected color of the bullets in both modes?

Screenshot of tutorial sidebar in light mode showing white bullet points

**Comment by Pyrax** [comment]
Thanks for your quick feedback! I changed the implementation details according to your suggestion.

**Comment by fk** [comment]
> In addition, I noticed that in dark mode, the bullets show up as white (the foreground color), while in light mode, they still show up as white (the background color). @fk, is this intentional? What's the expected color of the bullets in both modes?

Not intentional, … _sloppy_. All of this "step UI" is.
Let's do `ui.border` for now, and make those bullets a little more visible.

File: www/src/components/sidebar/item-link.js [hunk 1]
```diff
@@ -99,7 +99,7 @@ export default function ItemLink({ item, overrideCSS }) {
         onClick={onLinkClick}
         to={item.link}
       >
-        {item.ui === `steps` && (
+        {isSteps && (
           <span
             sx={{
               bg: `white`,
```
Comment (fk) [inline www/src/components/sidebar/item-link.js:105]: Hey @Pyrax! 👋 
Thanks so much for taking care of this! 🙏 

Could you change the `bg` to `ui.border` as mentioned in https://github.com/gatsbyjs/gatsby/pull/24129#issuecomment-630103040?

```suggestion
              bg: `ui.border`,
```

**Comment by fk** [comment]
🙏 Thanks you two!

**Pull request merged**